x-pack/libbeat/management/fleet/manager.go

```diff
@@ -0,0 +1,289 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package fleet
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/gofrs/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/common/reload"
+	"github.com/elastic/beats/libbeat/feature"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/management"
+	"github.com/elastic/beats/x-pack/libbeat/management/api"
+
+	xmanagement "github.com/elastic/beats/x-pack/libbeat/management"
+)
+
+var errEmptyAccessToken = errors.New("access_token is empty, you must re-enroll your Beat")
+
+// ConfigManager provides a functionality to retrieve config channel
+// using which manager is informed about config changes
+type ConfigManager interface {
+	ConfigChan() chan<- map[string]interface{}
+}
+
+func init() {
+	management.Register("x-pack-fleet", NewFleetManager, feature.Beta)
+}
+
+// Manager handles internal config updates. By retrieving
+// new configs from Kibana and applying them to the Beat
+type Manager struct {
+	config    *Config
+	cache     *xmanagement.Cache
+	logger    *logp.Logger
+	beatUUID  uuid.UUID
+	done      chan struct{}
+	registry  *reload.Registry
+	wg        sync.WaitGroup
+	blacklist *xmanagement.ConfigBlacklist
+
+	configChan chan map[string]interface{}
+}
+
+// NewFleetManager returns a X-Pack Beats Central Management manager
+func NewFleetManager(config *common.Config, registry *reload.Registry, beatUUID uuid.UUID) (management.ConfigManager, error) {
+	c := defaultConfig()
+	if config.Enabled() {
+		if err := config.Unpack(&c); err != nil {
+			return nil, errors.Wrap(err, "parsing central management settings")
+		}
+	}
+	return NewFleetManagerWithConfig(c, registry, beatUUID)
+}
+
+// NewFleetManagerWithConfig returns a X-Pack Beats Central Management manager
+func NewFleetManagerWithConfig(c *Config, registry *reload.Registry, beatUUID uuid.UUID) (management.ConfigManager, error) {
+	var cache *xmanagement.Cache
+	var blacklist *xmanagement.ConfigBlacklist
+
+	if c.Enabled && c.Mode == xmanagement.ModeFleet {
+		var err error
+
+		// Initialize configs blacklist
+		blacklist, err = xmanagement.NewConfigBlacklist(c.Blacklist)
+		if err != nil {
+			return nil, errors.Wrap(err, "wrong settings for configurations blacklist")
+		}
+
+		// Initialize central management settings cache
+		cache = &xmanagement.Cache{}
```

Comment:
I do not think we should have any caching done at the beats level, this could probably lead to inconsistency of how trhings are configured.